sktime/markov/hmm/init/discrete/_init_discrete_impl.py

```diff
@@ -0,0 +1,429 @@
+        sets. In case of '1/n', the threshold gets resolved to :math:`1 / \mathrm{n\_states\_full}`.
+
+    Returns
+    -------
+    hmm_init : HiddenMarkovStateModel
+        An initial guess for the HMM
+
+    See Also
+    --------
+    :class:`DiscreteOutputModel <sktime.markov.hmm.DiscreteOutputModel>`
+        The type of output model this heuristic uses.
+
+    :func:`metastable_from_msm`
+        Initial guess from an already existing :class:`MSM <sktime.markov.msm.MarkovStateModel>`.
+
+    :func:`init.gaussian.from_data <sktime.markov.hmm.init.gaussian.from_data>`
+        Initial guess with :class:`Gaussian output model <sktime.markov.hmm.GaussianOutputModel>`.
+
+
+    References
+    ----------
+    .. bibliography:: /references.bib
+        :style: unsrt
+        :filter: docname in docnames
+        :keyprefix: hmm-init-data-
+    """
+    if mode not in metastable_from_data.VALID_MODES \
+            + [m + "-regularized" for m in metastable_from_data.VALID_MODES]:
+        raise ValueError("mode can only be one of [{}]".format(", ".join(metastable_from_data.VALID_MODES)))
+
+    from sktime.util import ensure_dtraj_list
+    from sktime.markov.util import compute_dtrajs_effective
+    from sktime.markov import TransitionCountEstimator
+
+    dtrajs = ensure_dtraj_list(dtrajs)
+    dtrajs = compute_dtrajs_effective(dtrajs, lagtime=lagtime, n_states=n_hidden_states, stride=stride)
+    counts = TransitionCountEstimator(1, 'sliding', sparse=False).fit(dtrajs).fetch_model()
+    if states is not None:
+        counts = counts.submodel(states)
+    if '-regularized' in mode:
+        import msmtools.estimation as memest
+        counts.count_matrix[...] += memest.prior_neighbor(counts.count_matrix, 0.001)
+        nonempty = np.where(counts.count_matrix.sum(axis=0) + counts.count_matrix.sum(axis=1) > 0)[0]
+        counts.count_matrix[nonempty, nonempty] = np.maximum(counts.count_matrix[nonempty, nonempty], 0.001)
+    if 'all' in mode:
+        pass  # no-op
+    if 'largest' in mode:
+        counts = counts.submodel_largest(directed=True, connectivity_threshold=connectivity_threshold,
+                                         sort_by_population=False)
+    if 'populous' in mode:
+        counts = counts.submodel_largest(directed=True, connectivity_threshold=connectivity_threshold,
+                                         sort_by_population=True)
+    from sktime.markov.msm import MaximumLikelihoodMSM
+    msm = MaximumLikelihoodMSM(reversible=True, allow_disconnected=True, maxerr=1e-3,
+                               maxiter=10000).fit(counts).fetch_model()
+    return metastable_from_msm(msm, n_hidden_states, reversible, stationary, separate_symbols, regularize)
+
+
+metastable_from_data.VALID_MODES = ['all', 'largest', 'populous']
+
+
+def random(n_observation_states: int, n_hidden_states: int, seed: Optional[int] = None):
```

Comment:
fine if not or if this is too big a hassle but i wonder if this method name should be more descriptive? like more about what it does than just "random"?